packages/loader/container-loader/src/container.ts

```diff
@@ -1563,6 +1594,15 @@ export class Container
 
 				// eslint-disable-next-line @typescript-eslint/no-floating-promises
 				this._deltaManager.inbound.pause();
+
+				if (
+					loadMode.opsBeforeReturn === "sequenceNumber" &&
+					loadMode.freezeAfterLoad === true
+				) {
+					// Ensure this listener was turned off. It could technically be left on if the sequence number
+					// provided by the user has not been reached yet by any client.
+					this.off("op", freezeAtSeqNumListener);
```

Comment:
I'm not sure I understand the comment and semantics. The code above (waiting for ops) will give us some ops. There are not that many guarantees here - ops might be in transition and not be available through ordering service or storage for a moment. So we got _some_ ops. What's next?

Reply:
The idea here was to protect against a situation where a user requests the container to be frozen at a sequence number that does not exist yet. I suppose in this case we may want to throw an error though?

Reply:
As much as possible I'd prefer for these kinds of decisions not to be baked into Container/Loader. I.e. I'd rather provide a stand-alone API that has very specific behavior (like waits for sequence number, but throws if it's above it or we connected and still did not reach it). Such that someone else can easily build and use alternative implementation with different behavior. Baking it in Container layer will result eventually in 20 different options that all look the same but have slightly different behavior, and some of them can be chosen / specified at the same time (i.e. it's not enum, it's independent options with no clear description of how they interact with each other).